packages/flutter/lib/src/rendering/stack.dart

```diff
@@ -749,4 +749,20 @@ class RenderIndexedStack extends RenderStack {
     super.debugFillProperties(properties);
     properties.add(IntProperty('index', index));
   }
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    final List<DiagnosticsNode> children = <DiagnosticsNode>[];
+    int i = 0;
+    RenderObject? child = firstChild;
+    do {
```

Comment:
changes this to a `while` loop from a 
`do while` loop and you will avoid issues if child is null.